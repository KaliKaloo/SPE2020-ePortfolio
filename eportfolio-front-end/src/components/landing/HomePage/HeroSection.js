import React from 'react';
import './HeroSection.css';
import { Button } from './Button';
import { Link } from 'react-router-dom';


function HeroSection({
  lightBg,
  topLine,
  lightText,
  lightTextDesc,
  headline,
  description,
  buttontrue,
  buttonLabel,
  img,
  alt,
  imgStart,
}) {

  return (
    <>
      <div
        className={lightBg ? 'home__hero-section' : 'home__hero-section darkBg'}
      >
        <div className='container'>
          <div
            className='row home__hero-row'
            style={{
              display: 'flex',
              flexDirection: imgStart === 'start' ? 'row-reverse' : 'row'
            }}
          >
            <div className='col'>
              <div className='home__hero-text-wrapper'>
                <div className='top-line'>{topLine}</div>
                <div className={lightText ? 'heading' : 'heading dark'}>
                  {headline}
                </div>
                <p
                  className={
                    lightTextDesc
                      ? 'home__hero-subtitle'
                      : 'home__hero-subtitle dark'
                  }
                >
                  {description}
                </p>
                {buttontrue ? 
                    <>
                    <Link to='/register'>
                    <Button buttonSize='btn--wide' buttonColor='blue'>
                        {buttonLabel}
                    </Button>
                    </Link>
                    <Link to="/login" className={lightTextDesc ? 'login_text light':"login_text dark"}> Log in </Link>
                    </>
                    :
                    null
                }
              </div>
            </div>
            <div className='col'>
              <div className='home__hero-img-wrapper'>
                <img src={img} alt={alt} className='home__hero-img' />
              </div>
            </div>
          </div>
        </div>
      </div>
    </>
  );
}

export default HeroSection;
