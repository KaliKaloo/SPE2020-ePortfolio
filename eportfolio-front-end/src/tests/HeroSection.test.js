import React from 'react'
import { render, cleanup } from '@testing-library/react'
import '../../node_modules/bootstrap/dist/css/bootstrap.min.css';
import HeroSection from '../components/landing/HomePage/HeroSection';


afterEach(cleanup)

it('should take a snapshot for HeroSection UI', () => {
  const { asFragment } = render(<HeroSection />)

  expect(asFragment(<HeroSection />)).toMatchSnapshot()
})
