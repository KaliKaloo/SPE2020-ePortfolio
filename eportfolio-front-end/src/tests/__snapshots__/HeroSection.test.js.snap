// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot for HeroSection UI 1`] = `
<DocumentFragment>
  <div
    class="home__hero-section darkBg"
  >
    <div
      class="container"
    >
      <div
        class="row home__hero-row"
        style="display: flex; flex-direction: row;"
      >
        <div
          class="col"
        >
          <div
            class="home__hero-text-wrapper"
          >
            <div
              class="top-line"
            />
            <div
              class="heading dark"
            />
            <p
              class="home__hero-subtitle dark"
            />
          </div>
        </div>
        <div
          class="col"
        >
          <div
            class="home__hero-img-wrapper"
          >
            <img
              class="home__hero-img"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
